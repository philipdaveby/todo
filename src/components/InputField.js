import React, { useState } from 'react';
import PropTypes from 'prop-types';

const InputField = props => {
  const [title, setTitle] = useState('');
  const [content, setContent] = useState('');

  const constructTodo = e => {
    e.preventDefault();

    props.addTodo({
      title,
      content,
      done: false,
      key: props.todoKey,
    });
    setTitle('');
    setContent('');
  };

  return (
      <form aria-label="form" className="main__form__todo" onSubmit={ constructTodo }>
        <label className="main__form__label">
          Title
          <input type="text" name="Title" placeholder="Enter a title.." className="main__form__text-input" value={title} onChange={e => setTitle(e.target.value)} />
        </label>
        <label className="main__form__label">
          Description
          <input type="text" name="Description" placeholder="Enter a description.." className="main__form__text-input" value={content} onChange={e => setContent(e.target.value)} />
        </label>
        <input type="submit" id="input__submit" value="Add" />
      </form>
  );
};

InputField.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
};

export default InputField;
