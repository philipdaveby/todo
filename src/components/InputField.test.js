import InputField from './InputField';
import { render, screen, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('The InputField component', () => {

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<InputField />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a form tag', () => {
    render(<InputField />);
    const form = screen.getByRole('form');
    expect(form).toBeInTheDocument();
  });

  test('the form submits and calls addTodo', () => {
    const mock = jest.fn();
    const { getByRole } = render(<InputField addTodo={mock} todoKey={1} />);
    const form = getByRole('form');
    fireEvent.submit(form);
    expect(mock).toHaveBeenCalled();
  });

  test('Snapshot', () => {
    const inputComponent = renderer
      .create(<InputField todoKey={1} />)
      .toJSON();
    expect(inputComponent).toMatchSnapshot();
  });

});
