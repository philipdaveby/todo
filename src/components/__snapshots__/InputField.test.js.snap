// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The InputField component Snapshot 1`] = `
<form
  aria-label="form"
  className="main__form__todo"
  onSubmit={[Function]}
>
  <label
    className="main__form__label"
  >
    Title
    <input
      className="main__form__text-input"
      name="Title"
      onChange={[Function]}
      placeholder="Enter a title.."
      type="text"
      value=""
    />
  </label>
  <label
    className="main__form__label"
  >
    Description
    <input
      className="main__form__text-input"
      name="Description"
      onChange={[Function]}
      placeholder="Enter a description.."
      type="text"
      value=""
    />
  </label>
  <input
    id="input__submit"
    type="submit"
    value="Add"
  />
</form>
`;
